manufacturing-client/src/main.rs

```diff
@@ -291,15 +293,57 @@ async fn main() -> Result<()> {
 }
 
 async fn get_mfg_info(mfg_string_type: MfgStringType) -> Result<CborSimpleType> {
-    if let Some(mfg_info) = env::var_os("MANUFACTURING_INFO") {
```

Comment:
Well this is a bit knotty !  so this condition checks if mfs_info is present and equals to MANUFACTURING_INFO , then it sets the mfg_info as return value .This case is always true since we set that to   .env("MANUFACTURING_INFO", &mfg_args.device_info) in here -> https://github.com/fedora-iot/fido-device-onboard-rs/blob/main/admin-tool/src/aio/device.rs#L105. Therefore in OV Device_Info always reflects as the device name which passed in the command when device is created like target/debug/fdo-admin-tool aio --directory aio-dir device --device-credential-location testdevice1.dc manufacture --device-info testdevice1 . So if we keep this condition , testdevice1 appears as msf_info always. And never goes to check Serial Number (below match condition)
May be this needs some refactoring how the get_mfg_info() is written or a separate function for getting mfg_info and serialNumber.

Reply:
This only happens with the `aio` tool which is not used in production

Reply:
The developer has never had the OV with a "real" serial number, the env variable is there because we don't need anything real for testing how the servers work, which is the aim of `aio`. And also, we don't need to test everything with `aio`, it is mainly to test thing that require interactions with different servers.

I feel like you can handle this in two ways: 1) do not delete this environment variable, because it does not affect production and `aio` works fine the way that it is (its goal is to spin up the servers for testing quickly, so having a real device info is not relevant) and make a test outside e2e for your new functionality;
or 2) update *all* the e2e tests that require the handling of this environment variable, make your new tests (outside e2e) handling this new functionality, update the `aio` tool so that it can generate the Device Info in three different ways (1: serial number, 2: new behaviour with mac + serial number, 3: a way to overwrite 1 & 2 and simply put in the OV something that we want —which is what the env variable did—), and finally update the `aio` tool tests so that this new functionality of `aio` is tested.

Reply:
with the pov that "we don't need to test everything with aio, it is mainly to test thing that require interactions with different servers." I believe then it will be better to add a test outside e2e only to test new format of the serial number. (so that that avoids changes in all previous test cases and this new functionality gets tested anyways in separate test case). I am working on adding test case to this.